File: backend/app/routes/versions.py
from __future__ import annotations

from bson import ObjectId
from fastapi import APIRouter, Depends

from app.auth.dependencies import require_role
from app.database import get_db
from app.services.version_service import get_public_version_history

router = APIRouter(prefix="/api/v1/versions", tags=["versions"])


@router.get("/public/{project_id}")
async def public_history(project_id: str, db=Depends(get_db)):
    return {"data": await get_public_version_history(project_id, db), "error": None}


@router.get("/official/{project_id}")
async def official_history(project_id: str, user=Depends(require_role("official")), db=Depends(get_db)):
    return {"data": await get_public_version_history(project_id, db), "error": None}

File: frontend/src/api/versions.js
import client from "./client";

export const getPublicVersionHistory = async (projectId) => {
  const response = await client.get(`/api/v1/versions/public/${projectId}`);
  return response.data;
};

export const getOfficialVersionHistory = async (projectId) => {
  const response = await client.get(`/api/v1/versions/official/${projectId}`);
  return response.data;
};
